gym/core.py

```diff
@@ -325,8 +338,13 @@ def reset(self, **kwargs):
             return self.observation(self.env.reset(**kwargs))
 
     def step(self, action):
-        observation, reward, done, info = self.env.step(action)
-        return self.observation(observation), reward, done, info
+        step_returns = self.env.step(action)
+        if len(step_returns) == 5:
+            observation, reward, terminated, truncated, info = step_returns
+            return self.observation(observation), reward, terminated, truncated, info
+        else:
+            observation, reward, done, info = step_returns
+            return self.observation(observation), reward, done, info
```

Comment:
```suggestion
        observation, *rests = self.env.step(action)
        return (self.observation(observation), *rests)
```

Reply:
I kinda prefer the style suggested here, though the current version might be more simple to understand, so I think I'm fine with either

Reply:
I would prefer to keep this explicit, to show that this abstract class handles both APIs, especially since it's in the core file.